compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -573,6 +577,12 @@ public void visit(BLangTypeDefinition typeDefinition) {
             analyzeDef(typeDefinition.typeNode, env);
         }
 
+        if (typeDefinition.symbol == null) {
+            typeDefinition.symbol = Symbols.createTypeDefinitionSymbol(Flags.asMask(typeDefinition.flagSet),
+                    names.fromIdNode(typeDefinition.name), env.enclPkg.packageID, symTable.noType, env.scope.owner,
+                    typeDefinition.name.pos, getOrigin(typeDefinition.name.value));
+        }
+
```

Comment:
This looks hacky. Why do we need this? IIRC, to fix the issue, adding a null check in the symbol finder is enough.

Reply:
Without this, the table symbol will still be null and cannot perform symbol(), references() on it

Reply:
There's no point creating a type definition symbol if the type itself cannot be resolved, right? How is this handled for maps?

e.g.,
```ballerina
type TypeName map<>;
```

Didn't observe the NPE for this.

Reply:
Here we are not getting the NPE, but renaming and find references features are not working

Reply:
@mohanvive @nadeeshaan

Reply:
I believe @dulajdilshan 's idea is correct here. It is a common use-case that the user tends to find the references in these scenarios. Also, the go-to definition as well. Hence It is beneficial to handle this. +1

Reply:
As discussed (with @mohanvive @nadeeshaan and @pubudu91). This is a common cause https://github.com/ballerina-platform/ballerina-lang/issues/34706#issuecomment-1047592733 we have to handle generally. Therefore, we have decided to remove creating type-def symbol for this level.